### PR TITLE
Wifi config b1

### PR DESCRIPTION
**Pull request by MohammadAbbasi76**
add wifi component , that can get ssid and pass word from user 
wifi at first step go to AC mode and user can connect to wifi and 
write "wificonfig.local" and see setup page

### CODE REVIEW

**Review by Behn4m — COMMENTED** [review]
Thank you for the very clean PR, good job
The only point I want to mention is that I am not sure if it's a good idea to initiate WiFi in GlobalInit instead of the main. I think we need to discuss it in the next planning

File: components/SpotifyAuthorization/SpotifyAuthorization.c
Comment (ehsanziyaee) [inline components/SpotifyAuthorization/SpotifyAuthorization.c:1]: i think this method "esp_event_loop_create_default()" is real Global function and should be in Global Init (if not have any technical and compile-time error).

File: components/GlobalInit/CMakeLists.txt
Comment (ehsanziyaee) [inline components/GlobalInit/CMakeLists.txt:1]: cycle dependency !!! 
WifiComponent is better run in main with it's mother task (but should never occupied main heap mem. )

File: components/WiFiComponent/CMakeLists.txt
Comment (ehsanziyaee) [inline components/WiFiComponent/CMakeLists.txt:1]: delete last comment

**Review by ehsanziyaee — APPROVED** [review]

**Review by Behn4m — APPROVED** [review]

**Pull request merged**